src/main/webapp/components/query-selector/query-selector.js

```diff
@@ -12,31 +16,55 @@ import {
 } from '../index-cards'
 import AdvancedSearchQueryBuilder from '../query-builder/query-builder'
 const { useDrawInterface } = require('../../react-hooks')
+import useAnchorEl from '../../react-hooks/use-anchor-el'
 
-const useOpenClose = props => {
-  const [anchorEl, setAnchorEl] = React.useState(null)
-
-  const handleOpen = event => {
-    setAnchorEl(event.currentTarget)
-  }
+const defaultSearchForms = {
+  basic: BasicSearchQueryBuilder,
+  advanced: AdvancedSearchQueryBuilder,
+}
 
-  const handleClose = () => {
-    setAnchorEl(null)
-    if (props.onClose) {
-      props.onClose()
-    }
-  }
+const defaultSearchFormLabels = {
+  basic: 'Basic Search',
+  advanced: 'Advanced Search',
+}
 
-  const open = Boolean(anchorEl)
-  return [anchorEl, open, handleOpen, handleClose]
+const getBaseQueryFields = query => {
+  const { id, title, type } = query
+  return { id, title, type }
 }
 
+const defaultSearchFormInteractions = onChange => {
+  return Object.keys(defaultSearchForms).map(key => {
+    return (
+      <MenuItem key={key} onClick={() => onChange(key)}>
+        <SearchIcon style={{ marginRight: 10 }} />
+        <ListItemText>{defaultSearchFormLabels[key]}</ListItemText>
+      </MenuItem>
+    )
+  })
+}
 const QueryCard = props => {
-  const { onSearch, QueryEditor, onChange } = props
-  const [anchorEl, open, handleOpen, handleClose] = useOpenClose(props)
+  const { onSearch, QueryEditor, query } = props
+  const [anchorEl, handleOpen, handleClose, open] = useAnchorEl()
   const [{ active: isDrawing }] = useDrawInterface()
   const [wasDrawing, setWasDrawing] = useState(false)
   const drawAnchorEl = useRef(null)
+  const queryBuilder = defaultSearchForms.hasOwnProperty(query.type)
+    ? query.type
+    : 'advanced'
+
+  const onChange = query => {
+    props.onChange(query)
+  }
+
+  const queryInteractions = [
+    ...defaultSearchFormInteractions(key => {
+      if (key !== query.type) {
+        onChange(set(getBaseQueryFields(query), 'type', key))
+      }
+    }),
+  ]
+
```

Comment:
What is this doing?

Reply:
It resets the search form, but only does so if the search form type the user selects is different from the current one